### PR TITLE
Fix RandomCrop3D padding when transformation is skipped

### LINKED ISSUE #4654: RandomCrop3D returns the padded volume when gated off (p=0 with padding / pad_if_needed) — 3D twin of #772
## Summary
A gated-off `RandomCrop3D` with `padding` or `pad_if_needed` returns the **padded** volume rather than the input. This is the 3D twin of #772, which was fixed for the 2D class.

```python
import torch, kornia.augmentation as K
x = torch.arange(1, 121.).reshape(1, 1, 4, 5, 6)
K.RandomCrop3D((2, 3, 4), padding=2, p=0.)(x).shape            # (1, 1, 8, 9, 10)
K.RandomCrop3D((6, 7, 9), pad_if_needed=True, p=0.)(x).shape   # (1, 1, 8, 9, 12)
K.RandomCrop((2, 3), padding=2, p=0.)(torch.rand(1, 1, 5, 6)).shape   # (1, 1, 5, 6)  <- 2D is right
```

## Expected
`p=0` returns the input unchanged, as the 2D class does.

Measured on `main` @ c643312a3 merged with #4634 (docs-only), torch 2.14.0, CPU, macOS arm64. Found while reviewing #4634 (batch 6d of #4407).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

### PR DESCRIPTION
**Pull request by Junyi-Zheng**
## What does this pull request do?

RandomCrop3D previously padded the input before probability handling, so skipping the transformation could still change the output shape.

This fix moves padding into `apply_transform` and generates crop parameters using the padded shape. The skip path preserves the original input, while fixed and automatic padding retain their existing order.

Adds regression tests and a changelog entry.

## Related issue or discussion

Fixes #4654.

## What did you check?

Tested on macOS with Python 3.11.14 and PyTorch 2.14.0.

Before the fix, 17 regression cases failed with output shape mismatches. After the fix, the following suite reported 179 passed, 4 skipped, and 7 xfailed:

```bash
pixi run test-module tests/augmentation/_3d tests/augmentation/test_conventions_geometric_crop.py --device=cpu --dtype=float32,float64
```

The tests cover skipped transformations, input dimensionality, gradients, identity matrices, normal cropping, and parameter replay.

- `pixi run pre-commit-all` passed.
- `pixi run typecheck` completed successfully with one deprecation warning in the unchanged `lightglue.py` file.
- RandomCrop3D tests also passed on MPS with float32: 38 passed, 3 skipped, and 1 xfailed.
- CUDA was not tested locally.

## How did AI help?

AI helped clarify the issue in the initial stage.

### CODE REVIEW

**Review by ducha-aiki — APPROVED** [review]
Reviewed exact head 15f0ce985258e168f72024318fd9d9a1a6b6ced6.

Moving padding into apply_transform restores a genuinely untouched skip path, while forward_parameters still samples coordinates against the successively padded canvas. The padding-step computation preserves the previous ordering semantics, and the regressions cover explicit and automatic padding, dimensionality/keepdim, replay, gradients, transform identity, and multiple boundary modes. The full CI matrix is green.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-5).
